File: cdk_example/cdk_example_stack.ts
import { Stack, App, Duration } from 'aws-cdk-lib';
import { Ec2TaskDefinition, Cluster, ContainerImage, InstanceType, InstanceClass, InstanceSize } from 'aws-cdk-lib/aws-ecs';
import { Queue } from 'aws-cdk-lib/aws-sqs';
import { QueueProcessingEc2Service } from 'aws-cdk-lib/aws-ecs-patterns';
import { Construct } from 'constructs';
import { InstanceType } from 'aws-cdk-lib/aws-ec2';

export class FlaskSqsAppStack extends Stack {
    constructor(scope: Construct, id: string, props?: StackProps) {
        super(scope, id, props);

        // Define an SQS queue
        const queue = new Queue(this, "FlaskSqsQueue", {
            queueName: "FlaskSqsQueue",
            visibilityTimeout: Duration.seconds(300)
        });

        // Define an ECS cluster (where your service will be deployed)
        const cluster = new Cluster(this, "FlaskSqsAppCluster", {
            clusterName: "FlaskSqsAppCluster",
            defaultCloudMapNamespace: { name: "johns-name-space" }
        });

        const taskDefinition = new Ec2TaskDefinition(this, "QueueProcessingTaskDef");
        taskDefinition.addContainer("QueueProcessingContainer", {
            image: ContainerImage.fromRegistry("flask-sqs-app"),
            memoryLimitMiB: 512  // Specify the memory limit in MiB
        });

        cluster.addCapacity("CapacityId", {
            instanceType: new InstanceType(InstanceClass.BURSTABLE2, InstanceSize.MICRO),
            minCapacity: 1,
            maxCapacity: 3
        });

        // Create the ECS Service with the previously built Docker image
        new QueueProcessingEc2Service(this, "FlaskSqsService", {
            cluster: cluster,
            // update this to your Docker image URL
            image: ContainerImage.fromRegistry("740239033577.dkr.ecr.us-east-1.amazonaws.com/flask-sqs-app:latest"),
            minScalingCapacity: 1,
            environment: {
                QUEUE_NAME: queue.queueName
            },
            queue: queue,
            memoryLimitMiB: 512
        });
    }
}

const app = new App();
new FlaskSqsAppStack(app, "FlaskSqsAppStack");

File: cdk_example/cdk_example_stack.py
from aws_cdk import (
    Duration,
    Stack,
    App,
    aws_ec2 as ec2,
    aws_ecs as ecs, 
    aws_sqs as sqs,
)
from aws_cdk import aws_ecs_patterns

from constructs import Construct


class FlaskSqsAppStack(Stack):

    def __init__(self, scope: Construct, id: str, **kwargs) -> None:
        super().__init__(scope, id, **kwargs)

        # Define an SQS queue
        queue = sqs.Queue(
            self, "FlaskSqsQueue",
            queue_name="FlaskSqsQueue",
            visibility_timeout=Duration.seconds(300),
        )

        # Define an ECS cluster (where your service will be deployed)
        cluster = ecs.Cluster(
            self, "FlaskSqsAppCluster",
            cluster_name="FlaskSqsAppCluster",
            default_cloud_map_namespace=ecs.CloudMapNamespaceOptions(
                name="johns-name-space"
            ),            
        )
 
        task_definition = ecs.Ec2TaskDefinition(self, "QueueProcessingTaskDef")
        task_definition.add_container("QueueProcessingContainer",
            image=ecs.ContainerImage.from_registry("flask-sqs-app"),
            memory_limit_mib=512,  # Specify the memory limit in MiB
        )
        cluster.add_capacity(
            "CapacityId",
            instance_type=ec2.InstanceType.of(
                ec2.InstanceClass.BURSTABLE2, ec2.InstanceSize.MICRO
            ),
            min_capacity=1,
            max_capacity=3,
        )

        # # Create the ECS Service with the previously built Docker image
        aws_ecs_patterns.QueueProcessingEc2Service(
            self, "FlaskSqsService",
            cluster=cluster,
            image=ecs.ContainerImage.from_registry("740239033577.dkr.ecr.us-east-1.amazonaws.com/flask-sqs-app:latest"),
            min_scaling_capacity=1,
            environment={
                "QUEUE_NAME": queue.queue_name
            },
            queue=queue,
            memory_limit_mib=512,  # Specify the memory limit in MiB
        )

app = App()
FlaskSqsAppStack(app, "FlaskSqsAppStack")
